src/main/scala/cognite/spark/v1/DefaultSource.scala

```diff
@@ -85,13 +94,114 @@ class DefaultSource
     new SequenceRowsRelation(config, sequenceId)(sqlContext)
   }
 
+  private def getFunctionResult(
+      client: GenericClient[IO],
+      functionId: Long,
+      result: FunctionCall): IO[Json] = {
+    var res = result
+    while (res.status != None && res.status.get == "Running") {
```

Comment:
This is potentially infinite, maybe we can use `retryWhile` here?

Reply:
I could not find a `retryWhile` function. Instead I added a counter. 30 retries seem about right from what I have experienced in datastudio.

Reply:
It's this one: https://github.com/cognitedata/cdp-spark-datasource/blob/3db8b5401685e4d9736c9bd6fb03196fef43e3f7/src/test/scala/cognite/spark/v1/SparkTest.scala#L72
For this use case we probably wouldn't want an exponential backoff. Either way, we can change that later.